src/Dashboard/Grid.php

```diff
@@ -1189,266 +1159,302 @@ public function getAllDasboardCards($force = false): array
-                    'user_observer'   => __("Top ticket's observers"),
-                    'group_observer'  => __("Top ticket's observer groups"),
-                    'user_assign'     => __("Top ticket's assignees"),
-                    'group_assign'    => __("Top ticket's assignee groups"),
-                ] as $type => $label
-            ) {
-                $cards["top_ticket_$type"] = [
-                    'widgettype' => ['pie', 'donut', 'halfpie', 'halfdonut', 'summaryNumbers', 'multipleNumber', 'bar', 'hbar'],
-                    'itemtype'   => "\\Ticket",
-                    'group'      => __('Assistance'),
-                    'label'      => $label,
-                    'provider'   => "Glpi\\Dashboard\\Provider::nbTicketsActor",
-                    'args'       => [
-                        'case' => $type,
-                    ],
-                    'filters'    => [
-                        'dates', 'dates_mod', 'itilcategory',
-                        'group_tech', 'user_tech', 'requesttype', 'location'
-                    ]
-                ];
-            }
-
-            $cards["RemindersList"] = [
-                'widgettype' => ["articleList"],
-                'label'      => __("List of reminders"),
-                'group'      => __('Tools'),
-                'provider'   => "Glpi\\Dashboard\\Provider::getArticleListReminder",
-                'filters'    => ['dates', 'dates_mod']
-            ];
+        $cards["RemindersList"] = [
+            'widgettype' => ["articleList"],
+            'label'      => __("List of reminders"),
+            'group'      => __('Tools'),
+            'provider'   => "Glpi\\Dashboard\\Provider::getArticleListReminder",
+            'filters'    => ['dates', 'dates_mod']
+        ];
 
-            $cards["markdown_editable"] = [
-                'widgettype'   => ["markdown"],
-                'label'        => __("Editable markdown card"),
-                'group'        => __('Others'),
-                'card_options' => [
-                    'content' => __("Toggle edit mode to edit content"),
-                ]
-            ];
+        $cards["markdown_editable"] = [
+            'widgettype'   => ["markdown"],
+            'label'        => __("Editable markdown card"),
+            'group'        => __('Others'),
+            'card_options' => [
+                'content' => __("Toggle edit mode to edit content"),
+            ]
+        ];
 
-            $more_cards = Plugin::doHookFunction(Hooks::DASHBOARD_CARDS);
-            if (is_array($more_cards)) {
-                $cards = array_merge($cards, $more_cards);
-            }
+        $more_cards = Plugin::doHookFunction(Hooks::DASHBOARD_CARDS);
+        if (is_array($more_cards)) {
+            $cards = array_merge($cards, $more_cards);
         }
 
+        // Set cards specifications to cache
+        $GLPI_CACHE->set('getAllDasboardCards', $cards, new DateInterval('PT5M'));
```

Comment:
Why only 5 minutes ?

Reply:
The cache will be cleared on each reload anyway (see other comments above) so I don't think the 5 minute limit will be reached.

If we "break" plugin compatibility in 10.1 and remove the cache clear on page load then 5 min seems a fair limit for unexpected cases (e.g. someone running a patch that add a new card type and might no think of running the clear cache command).
This cache save about 0.2s of computations, which is a big deal in multiple simultaneous AJAX calls but shouldn't hurt once every 5 min.

Reply:
As discussion, changed to 1 day in the last commit.